python/pyspark/sql/types.py

```diff
@@ -1299,6 +1300,12 @@ def fromInternal(self, obj: Tuple) -> "Row":
         return _create_row(self.names, values)
 
 
+class VariantType(AtomicType):
+    """Variant data type, representing semi-structured values."""
```

Comment:
nit: if it's not for internal use only, shall we add `versionadded`?

Reply:
sounds good to me!